sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -400,6 +406,26 @@ class kernel_bundle : public detail::kernel_bundle_plain,
     return detail::kernel_bundle_plain::ext_oneapi_has_kernel(name);
   }
 
+  // For free functions
+  template <auto *Func>
+  std::enable_if_t<ext::oneapi::experimental::is_kernel_v<Func>, bool>
+  ext_oneapi_has_kernel() {
+    return has_kernel(ext::oneapi::experimental::get_kernel_id<Func>());
+  }
+
+  template <auto *Func>
+  std::enable_if_t<ext::oneapi::experimental::is_kernel_v<Func>, bool>
+  ext_oneapi_has_kernel(const device &dev) {
+    return has_kernel(ext::oneapi::experimental::get_kernel_id<Func>(), dev);
+  }
+
+  template <auto *Func>
+  std::enable_if_t<ext::oneapi::experimental::is_kernel_v<Func>, kernel>
```

Comment:
This also needs to be constrained on the bundle state:

> _Constraints:_ Available only if State is `bundle_state::executable` and if `is_kernel_v<Func>` is `true`.

You can see some examples above for constraining a member function on the bundle state.

Reply:
Added constraint on bundle_state.